src/test-table.md

```diff
@@ -100,6 +104,75 @@ for _, tt := range tests {
 }
 ```
 
+### Avoid Messy Table Tests
+
+Table tests become difficult to read and maintain if there becomes significant
+logic forks within the subtest `for` loop. Table tests should **NOT** be used whenever 
+there needs to be complex logic added inside this execution loop.
+
+Large complex table tests harm readability and maintainability because test readers will
+have difficulty debugging test failures that occur.
+
+Table tests like this should be split into either multiple test tables or multiple 
+individual `Test...` functions
+
+Some general guidelines to follow are:
```

Comment:
Suggestion:
```suggestion
Some ideals to aim for are:
```

That makes it clear that the "perfect test" will meet the following conditions but there's leeway as the authors deem fit.